sdk/turing/session.py

```diff
@@ -0,0 +1,162 @@
+    def active_project(self) -> Optional[Project]:
+        return self._project
+
+    @active_project.setter
+    def active_project(self, project):
+        mlflow.tracking.set_tracking_uri(project.mlflow_tracking_url)
+        self._project = project
+
+    def set_project(self, project_name: str):
+        """
+        Set this session's active projects
+        """
+        self.active_project = self.get_project_by_name(project_name)
+
+    def list_projects(self, name: Optional[str] = None) -> List[Project]:
+        """
+        List all projects, that the current user has access to
+
+        :param name: filter projects by name
+        :return: list of projects
+        """
+        kwargs = {}
+        if name:
+            kwargs["name"] = name
+        return ProjectApi(self._api_client).projects_get(**kwargs)
+
+    def get_project_by_name(self, project_name: str) -> Project:
+        """
+        Get MLP project by its name
+
+        :param project_name: name of the project
+        :raise Exception if the project with given name doesn't exist
+        :return: Project details
+        """
+        p_list = self.list_projects(name=project_name)
+
+        filtered = [p for p in p_list if p.name == project_name][:1]
+        if not filtered:
+            raise Exception(
+                f"{project_name} does not exist or you don't have access to the project. Please create new "
+                f"project using MLP console or ask the project's administrator to be able to access "
+                f"existing project.")
+
+        return filtered[0]
+
+    @require_active_project
+    def list_ensemblers(
+            self,
+            ensembler_type: Optional[EnsemblerType] = None,
+            page: Optional[int] = None,
+            page_size: Optional[int] = None) -> EnsemblersPaginatedResults:
+        """
+        List ensemblers
+        """
+        kwargs = {}
+
+        if ensembler_type:
+            kwargs["type"] = ensembler_type.value
+        if page:
+            kwargs["page"] = page
+        if page_size:
+            kwargs["page_size"] = page_size
+
+        return EnsemblerApi(self._api_client).list_ensemblers(
+            project_id=self.active_project.id,
+            **kwargs
+        )
+
+    @require_active_project
+    def create_ensembler(self, ensembler: Ensembler) -> Ensembler:
+        """
+        Create a new ensembler in the Turing back-end
+        """
+        return EnsemblerApi(self._api_client).create_ensembler(
+            project_id=self.active_project.id,
+            ensembler=ensembler)
+
+    @require_active_project
+    def update_ensembler(self, ensembler: Ensembler) -> Ensembler:
+        """
+        Update existing ensembler
+        """
+        return EnsemblerApi(self._api_client).update_ensembler(
+            project_id=ensembler.project_id,
+            ensembler_id=ensembler.id,
+            ensembler=ensembler)
+
+
+active_session: TuringSession = TuringSession(
```

Comment:
what's this for?

Reply:
To store the information about what project the user is working under and to provide the bridge between SDK methods, exposed to the user, and API endpoints

Reply:
But what if the turing is not hosted on the user's machine? Importing this file will cause this to run and throw an exception?

Reply:
It's just a default session, it's expected that user will initialize it with a proper Turing API before using:
```python
turing.set_url("http://turing.com/api", use_google_oauth=True)
```

If user has skipped this step, then the error will only occur when user will try to call Turing API with one of available methods, for instance `turing.Project.list()`. The error is not being thrown just on import.